src/client.ts

```diff
@@ -355,6 +390,69 @@ export class Client {
     }
   }
 
+  /**
+   * This Method allows the user to take a screenshot. If no selector is passed
+   * then a full page screenshot is saved.
+   * if no filename is passed, then file is stored with a timestamp as the filename
+   * @param params filename, selector
+   * @returns Filename
+   */
+  public async takeScreenshot(
+    params?: { fileName?: string; selector?: string },
+  ): Promise<string> {
+    if (!this.screenshot_folder || !existsSync(this.screenshot_folder)) {
+      throw new Error("The Screenshot folder is not set or doesn't exist");
+    }
+
+    let fileName = `${generateTimestamp()}.jpg`;
+    const options: SSOptions = {
+      format: this.screenshot_format,
+      quality: this.screenshot_quality,
+    };
+    if (params) {
+      if (params.fileName) {
+        fileName = params.fileName + ".jpg";
+      }
+      if (params.selector) {
+        const viewPort = await this.getViewport(params.selector);
+        options.clip = viewPort;
+      }
+    }
+    const res = await this.sendWebSocketMessage(
+      "Page.captureScreenshot",
+      {
+        format: options.format,
+        quality: options.quality,
+        clip: options.clip,
+      },
+    ) as {
+      data: string;
+    };
+
+    //Writing the Obtained Base64 encoded string to image file
+    fileName = `${this.screenshot_folder}/${fileName}`;
+    const B64str = (res as { data: string }).data;
+    const u8Arr = Uint8Array.from<string>(atob(B64str), (c) => c.charCodeAt(0));
+    try {
+      Deno.writeFileSync(fileName, u8Arr);
+    } catch (e) {
+      console.error(
+        "Write Image to File Failed. Please check FileName and Path",
+      );
+      throw e;
+    }
+
+    return fileName;
+  }
+  /**
+   * To set the Folder for screenshots,
+   * so that we can save specific screenshots in specific folders.
+   * Folders need to be present already.
+   * @param FolderPath
+   */
+  public setScreenshotsFolder(FolderPath: string) {
+    this.screenshot_folder = FolderPath;
+  }
```

Comment:
I don't think we should need this as a user can just path in the direct path to the takeScreenshot method